lib/food_order.rb

```diff
@@ -0,0 +1,30 @@
+class FoodOrder
+  attr_accessor :basket, :takeaway_menu, :total_order
+  attr_reader :message
+
+  def initialize(takeaway_menu = Menu.new, message = TextClient.new)
+    @basket = []
+    @takeaway_menu = takeaway_menu
+    @message = message
+    @total_order = 0
+  end
+
+  def add_dish(dish)
+    selection = takeaway_menu.menu.select { |key, _value| key == dish }
+    basket << selection
+  end
+
+  def total
+    basket.each { |item| item.each_value { |cost| self.total_order += cost } }
+    "Your total order is £%0.2f" % [total_order]
+  end
```

Comment:
I would avoid storing the total in an instance variable (remove `@total_order`). Storing it causes duplicated information. If you have a basket stored already, the total can always be calculated on the fly based on that basket. Not storing duplicated information avoids bugs when information is out of sync (eg the user expects the order_total to be up to date once they've added something to their basket but it is not). 
Make total_order be a method that returns the calculated total everytime.